From 327e0630d3733888cff48a71866acba0e2544c54 Mon Sep 17 00:00:00 2001
From: capezotte <carana2099@gmail.com>
Date: Sat, 22 Oct 2022 13:06:46 -0300
Subject: [PATCH 2/3] Musl stdout fix

---
 src/SneedacityApp.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/SneedacityApp.cpp b/src/SneedacityApp.cpp
index 4fb5d75eb..431ebc68f 100644
--- a/src/SneedacityApp.cpp
+++ b/src/SneedacityApp.cpp
@@ -747,8 +747,8 @@ int main(int argc, char *argv[])
    // never be able to get rid of the messages entirely, but we should
    // look into what's causing them, so allow them to show in Debug
    // builds.
-   stdout = freopen("/dev/null", "w", stdout);
-   stderr = freopen("/dev/null", "w", stderr);
+   freopen("/dev/null", "w", stdout);
+   freopen("/dev/null", "w", stderr);
 
    return wxEntry(argc, argv);
 }
-- 
2.38.1

